common/ringpop/ringpop_test.go

```diff
@@ -26,29 +26,108 @@ package ringpop
 
 import (
 	"context"
+	"crypto/tls"
 	"fmt"
+	"os"
 	"testing"
 	"time"
 
 	"go.temporal.io/server/common/config"
+	"go.temporal.io/server/common/dynamicconfig"
 	"go.temporal.io/server/common/log"
+	"go.temporal.io/server/common/rpc/encryption"
+	"go.temporal.io/server/tests/testhelper"
 
+	"github.com/golang/mock/gomock"
 	"github.com/stretchr/testify/require"
 	"github.com/stretchr/testify/suite"
 	"gopkg.in/yaml.v3"
 )
 
-type RingpopSuite struct {
-	*require.Assertions
-	suite.Suite
-}
+type (
+	RingpopSuite struct {
+		*require.Assertions
+		suite.Suite
+
+		controller *gomock.Controller
+
+		logger           log.Logger
+		internodeCertDir string
+		internodeChain   testhelper.CertChain
+
+		membershipConfig         config.Membership
+		internodeConfigMutualTLS config.GroupTLS
+		internodeConfigServerTLS config.GroupTLS
+
+		ringpopMutualTLSFactoryA *ringpopFactory
+		ringpopMutualTLSFactoryB *ringpopFactory
+		ringpopServerTLSFactoryA *ringpopFactory
+		ringpopServerTLSFactoryB *ringpopFactory
+
+		insecureFactory *ringpopFactory
+	}
+)
+
+const (
+	localhostIPv4 = "127.0.0.1"
+)
+
+var (
+	rpcTestCfgDefault = &config.RPC{
+		GRPCPort:       0,
+		MembershipPort: 7600,
```

Comment:
does this require the port 7600 to be available for the test to pass?

Reply:
It does, but this was just a relocated test.

Previously this was under the rpc package.